src/main/java/io/reactivex/internal/operators/observable/ObservableSwitchMap.java

```diff
@@ -131,15 +131,15 @@ public void onNext(T t) {
 
         @Override
         public void onError(Throwable t) {
-            if (done || !errors.addThrowable(t)) {
+            if (!done && errors.addThrowable(t)) {
```

Comment:
Codecov indicates that some of possible variants are not tested, do you want to cover them in this PR?

<img width="602" alt="screen shot 2018-02-01 at 4 48 30 pm" src="https://user-images.githubusercontent.com/967132/35711078-db98f0d0-076f-11e8-883b-61c3f6fb6b2f.png">

Reply:
It's not that I think code in the patch is broken, but that if we had test that was trying to go into `onError` when it's already `done == true`, we would have a hint that original code had some problems

Reply:
That line is tested but parallelism on the CI is flaky and may not produce the necessary interleaving there.

The upstream may onError after a crash in the mapper thus we have to route that into the global error handler.

Reply:
Can't we test it just by breaking RS protocol with  `main.onComplete(); main.onError()`?

I might miss something, reviewing on mobile

Reply:
[This](https://github.com/akarnokd/RxJava/blob/8ec030fc052e20b62da80bc6edf6d084c03094d8/src/test/java/io/reactivex/internal/operators/flowable/FlowableSwitchTest.java#L870) and [this](https://github.com/akarnokd/RxJava/blob/8ec030fc052e20b62da80bc6edf6d084c03094d8/src/test/java/io/reactivex/internal/operators/flowable/FlowableSwitchTest.java#L1002) tests should cover that line. Apparently, the test did not involve ps2 so the inner source was not involved. Updated the unit tests accordingly.